core/src/main/java/fr/sncf/osrd/envelope_sim/allowances/AbstractAllowanceWithRanges.java

```diff
@@ -309,22 +307,55 @@ public Envelope computeIteration(Envelope base,
 
         // 1) compute the potential junction parts (slowdown or speedup)
         var leftPart = computeLeftJunction(base, coreEnvelope, imposedBeginSpeed);
+
         var leftPartEndPos = leftPart != null ? leftPart.getEndPos() : base.getBeginPos();
         var coreEnvelopeWithLeft = computeEnvelopeWithLeftJunction(base, coreEnvelope, leftPart);
 
         var rightPart = computeRightJunction(base, coreEnvelopeWithLeft, imposedEndSpeed);
         var rightPartBeginPos = rightPart != null ? rightPart.getBeginPos() : base.getEndPos();
 
-        // if the junction parts touch or intersect, there is no core phase
-        if (rightPartBeginPos <= leftPartEndPos) {
+        if (EnvelopePhysics.areIntersecting(rightPart, leftPart)) {
+            // if the junction parts touch or intersect, there is no core phase
             return intersectLeftRightParts(leftPart, rightPart);
+        } else if (rightPartBeginPos < leftPartEndPos)
+            // if the junction parts neither touch nor intersect but their positions overlap,
+            // it is physically impossible to meet the two imposed speeds
+            throw tooMuchTime();
+
+        if (leftPart != null
+                && coreEnvelope.stream().noneMatch(part -> EnvelopePhysics.areIntersecting(part, leftPart))) {
+            // The slowdown section doesn't intersect with the core part
+            // That may happen when the section is too short or consists exclusively of braking / acceleration parts
+            throw tooMuchTime();
+        }
+        if (rightPart != null
+                && coreEnvelope.stream().noneMatch(part -> EnvelopePhysics.areIntersecting(part, rightPart))) {
+            // The speedup section doesn't intersect with the core part
+            // That may happen when the section is too short or consists exclusively of braking / acceleration parts
+            throw tooMuchTime();
```

Comment:
I don't get it, the speedup phase is computed with the code phase as a ceiling. does this code detect when the speedup / slowdown never intersect with the core phase, but rather with the end of the region? if so, this data is very available in a simpler way when computing these envelopes in the first place

Reply:
I changed it to `if (leftPart != null && coreEnvelope.getEndPos() == leftPartEndPos)`, is that what you had in mind or is there something simpler?